lib/saml_idp/request.rb

```diff
@@ -170,12 +170,34 @@ def matching_cert
 
       Array(service_provider.certs).find do |cert|
         document.valid_signature?(
-          OpenSSL::Digest::SHA256.new(cert.to_der).hexdigest,
+          fingerprint(cert),
           options.merge(cert:, digest_method_fix_enabled: true)
         )
       end
     end
 
+    def cert_errors
+      return nil unless signed?
+      begin
+        return nil if matching_cert.present?
+      rescue SamlIdp::XMLSecurity::SignedDocument::ValidationError => e
```

Comment:
this is necessary in case the SAML request has a certificate node but the node is empty. (this is the only ValidationError that is currently being bubbled up to the IdP. we could consider changing that, but i was trying to minimize my impact on existing code for this PR. thoughts about that welcome!)